web/pgadmin/tools/sqleditor/static/js/components/QueryToolComponent.jsx

```diff
@@ -226,11 +226,12 @@ export default function QueryToolComponent({params, pgWindow, pgAdmin, selectedN
     pollTime = -1;
   }
 
-  const handleEndOfLineChange = useCallback((e)=>{
+  const handleEndOfLineChange = useCallback((e, queryChanged=true)=>{
     const val = e.value || e;
```

Comment:
Yes, you are right it calls only once and I provided this queryChanged param to prevent the save button from enabling when we load the new file.
The code mirror is not detecting the EOL character because we are not changing the editor's text using the lineSeparator() (codemirror method).
We are just taking care of this EOL character in 2 cases:-
download/save and copy.

Reply:
For that reason we have CustomEditorView to implement our own behaviour. EOL feature should be part of CodeMirror. Query tool can only change it.